FastSimulation/TrackingRecHitProducer/python/TrackingRecHitProducer_cfi.py

```diff
@@ -1,58 +1,33 @@
+# Python 2 vs 3 compatibility library:
+import six
+
 import FWCore.ParameterSet.Config as cms
 
-import six
 
+# Load the detailed configurations of pixel plugins.
+# NB: for any new detector geometry (e.g. Phase 2 varians), we should write a new plugin
+# config file, and import it here, and below use its own Era to load it.
+#
+from FastSimulation.TrackingRecHitProducer.PixelPluginsPhase0_cfi import pixelPluginsPhase0
+from FastSimulation.TrackingRecHitProducer.PixelPluginsPhase1_cfi import pixelPluginsPhase1
+from FastSimulation.TrackingRecHitProducer.PixelPluginsPhase2_cfi import pixelPluginsPhase2
+
+# The default is (for better of worse) Phase 0:
+#
 fastTrackerRecHits = cms.EDProducer("TrackingRecHitProducer",
     simHits = cms.InputTag("fastSimProducer","TrackerHits"),
-    plugins=cms.VPSet()
+    plugins = pixelPluginsPhase0
 )
 
-fastTrackerRecHits.plugins.append(
-    cms.PSet(
-        name = cms.string("pixelBarrelSmearer"),
-        type=cms.string("PixelBarrelTemplateSmearerPlugin"),
-        BigPixelBarrelResolutionFile = cms.string('FastSimulation/TrackingRecHitProducer/data/NewPixelResolutionBarrel38T.root'),
-        EdgePixelBarrelResolutionFile = cms.string('FastSimulation/TrackingRecHitProducer/data/NewPixelResolutionBarrelEdge38T.root'),
-        RegularPixelBarrelResolutionFile = cms.string('FastSimulation/TrackingRecHitProducer/data/PixelBarrelResolution2014.root'),
-        BigPixelForwardResolutionFile = cms.string('FastSimulation/TrackingRecHitProducer/data/NewPixelResolutionForward38T.root'),
-        EdgePixelForwardResolutionFile = cms.string('FastSimulation/TrackingRecHitProducer/data/NewPixelResolutionForward38T.root'),
-        RegularPixelForwardResolutionFile = cms.string('FastSimulation/TrackingRecHitProducer/data/PixelForwardResolution2014.root'),
-        UseCMSSWPixelParametrization = cms.bool(False),
-        MergeHitsOn = cms.bool(False),
-        MergingProbabilityBarrelFile = cms.string('FastSimulation/TrackingRecHitProducer/data/bmergeprob.root'),
-        MergingProbabilityForwardFile = cms.string('FastSimulation/TrackingRecHitProducer/data/fmergeprob.root'),
-        MergedPixelBarrelResolutionXFile = cms.string('FastSimulation/TrackingRecHitProducer/data/bxsmear.root'),
-        MergedPixelForwardResolutionXFile = cms.string('FastSimulation/TrackingRecHitProducer/data/fxsmear.root'),
-        MergedPixelBarrelResolutionYFile = cms.string('FastSimulation/TrackingRecHitProducer/data/bysmear.root'),
-        MergedPixelForwardResolutionYFile = cms.string('FastSimulation/TrackingRecHitProducer/data/fysmear.root'),
-        templateId = cms.int32( 40 ),
-        select=cms.string("subdetId==BPX"),
-    )
-)
+# Phase 1 Era: replace plugins by Phase 1 plugins
+from Configuration.Eras.Modifier_trackingPhase1_cff import trackingPhase1
+trackingPhase1.toModify(fastTrackerRecHits, plugins = pixelPluginsPhase1)
```

Comment:
These are properties of the detector, right? If yes, please use `phase1Pixel` modifier instead. The tracking-one is used to control the tracking configuration, which, to some degree, can be independent of the detector.